src/Resolver/File.php

```diff
@@ -81,6 +81,16 @@ public function resolve($definition)
             $path = realpath($this->getIterator()->getRootDefinition()->getBasepath() . \DIRECTORY_SEPARATOR . $path);
         }
 
-        return file_get_contents($path);
+        $content = file_get_contents($path);
+
+        if (pathinfo($path, PATHINFO_EXTENSION) == 'json') {
+            $content = json_decode($content, true);
+
+            if (json_last_error() !== JSON_ERROR_NONE) {
+                throw new \RuntimeException('Failed to parse ' . basename($path) . ': ' . json_last_error_msg());
```

Comment:
Just a comment, `upward-js` doesn't have json validation. Should we suggest it over on magento-research/pwa-studio#812?

Reply:
Are you sure `JSON.parse()` doesn't already throw an exception/error for invalid syntax? PHP's the difficult on in this case.

Reply:
Ah, yes; JS does throw a SyntaxError.